apps/site/src/app/schedule/components/ShiftingCountdown/ShiftingCountdown.tsx

```diff
@@ -0,0 +1,151 @@
+"use client";
+
+import { AnimatePresence, motion } from "framer-motion";
+import { useEffect, useRef, useState } from "react";
+import BarLoader from "../BarLoader";
+import clsx from "clsx";
+import { Fireworks } from "@fireworks-js/react";
+
+import styles from "./ShiftingCountdown.module.scss";
+
+// NOTE: Change this date to whatever date you want to countdown to :)
+// const HACKING_STARTS = "2024-01-26T18:00:00.000";
+const HACKING_ENDS = "2024-01-28T24:00:00.000";
+const HACKING_STARTS = "2023-12-06T22:23:00.000"; // Test when countdown ends
+// HACKING_STARTS.setSeconds(10);
+// const HACKING_ENDS = new Date();
+
+const SECOND = 1000;
+const MINUTE = SECOND * 60;
+const HOUR = MINUTE * 60;
+const DAY = HOUR * 24;
+
+const ShiftingCountdown = () => {
+	const intervalRef = useRef<ReturnType<typeof setInterval> | null>(null);
+	const [showTimer, setShowTimer] = useState(false);
+	const [showFireworks, setShowFireworks] = useState(false);
+	const [timerLabel, setTimerLabel] = useState("Hacking Starts");
```

Comment:
Suggestion: derived states make the logic more difficult to track. I think for all of these, we could compute the resultant values from the remaining time rather than maintaining separate states for them.

Reply:
I agree separate states are difficult to track. Though wouldn't I need state changes to perform a re-render in psuedo-real-time? How do you suggest I implement it?

Reply:
A re-render will happen whenever any state is updated, and you have the countdown's `remaining` state which is constantly updated.